sense-android-library/src/nl/sense_os/datastorageengine/RealmSensor.java

```diff
@@ -0,0 +1,234 @@
+     * @param value
+     * @param date
+     */
+    public void insertDataPoint(boolean value, long date) {
+        insertDataPoint(new DataPoint(id, value, date));
+    }
+
+    /**
+     * Insert a new DataPoint to this sensor
+     * @param value
+     * @param date
+     */
+    public void insertDataPoint(float value, long date) {
+        insertDataPoint(new DataPoint(id, value, date));
+    }
+
+    /**
+     * Insert a new DataPoint to this sensor
+     * @param value
+     * @param date
+     */
+    public void insertDataPoint(int value, long date) {
+        insertDataPoint(new DataPoint(id, value, date));
+    }
+
+    /**
+     * Insert a new DataPoint to this sensor
+     * @param value
+     * @param date
+     */
+    public void insertDataPoint(JSONObject value, long date) {
+        insertDataPoint(new DataPoint(id, value, date));
+    }
+
+    /**
+     * Insert a new DataPoint to this sensor
+     * @param value
+     * @param date
+     */
+    public void insertDataPoint(String value, long date) {
+        insertDataPoint(new DataPoint(id, value, date));
+    }
+
+    /**
+     * Update RealmSensor in local database with the info of the given Sensor object. Throws an exception if it fails to updated.
+     */
+    protected void saveChanges() throws DatabaseHandlerException {
+        realm.beginTransaction();
+
+        // get the existing sensor (just to throw an error if it doesn't exist)
+        RealmModelSensor realmSensor = realm
+                .where(RealmModelSensor.class)
+                .equalTo("id", this.id)
+                .findFirst();
+
+        if (realmSensor == null) {
+            throw new DatabaseHandlerException("Cannot update sensor: sensor doesn't yet exist.");
+        }
+
+        // update the sensor
+        realm.copyToRealmOrUpdate(RealmModelSensor.fromSensor(this));
+
+        realm.commitTransaction();
+    }
+
+    /**
+     * This method inserts a DataPoint object into the local Realm database.
+     * The typecasting to string should already be done at this point.
+     * @param dataPoint	A DataPoint object that has a stringified value that will be copied
+     * 			into a Realm object.
+     */
+    protected void insertDataPoint (DataPoint dataPoint) {
+        RealmModelDataPoint realmDataPoint = RealmModelDataPoint.fromDataPoint(dataPoint);
+
+        realm.beginTransaction();
+        realm.copyToRealmOrUpdate(realmDataPoint);
+        realm.commitTransaction();
+    }
+
+    /**
+     * Get data points from this sensor from the local database
+     * @param startDate: Start date of the query, included.
+     * @param endDate: End date of the query, excluded.
+     * @param limit: The maximum number of data points.
+     * @param sortOrder: Sort order, either ASC or DESC
+     * @return Returns a List with data points
+     */
+    public List<DataPoint> getDataPoints(long startDate, long endDate, int limit, DatabaseHandler.SORT_ORDER sortOrder) throws JSONException {
```

Comment:
In iOS, there is a check for invalid limit value.If it is equal or less than 0, then an InvalidLimit exception. Don't know if it is necessary. Otherwise, 0 will always return for an invalid limit.